### ISSUE TITLE
Subtitles not working properly (region and timing)

### ISSUE THREAD

**Issue by dommerq** [google/shaka-player#2157]
**Have you read the [FAQ](https://bit.ly/ShakaFAQ) and checked for duplicate open issues?**
Yes

**What version of Shaka Player are you using?**
Latest (2.5.5) have the issue as well

**Can you reproduce the issue with our latest release version?**
Yes

**Can you reproduce the issue with the latest code from `master`?**
Yes

**Are you using the demo app or your own custom app?**
Custom app

**If custom app, can you reproduce the issue using our demo app?**
Yes

**What browser and OS are you using?**
Chrome os (via caf chromecast but also happening on OSX/chrome)

**For embedded devices (smart TVs, etc.), what model and firmware version are you using?**
CAF on chromecast

**What are the manifest and license server URIs?**
Will be sent by email, including this issue number

**What did you do?**
Play, enable subtitles to italiano, watch at 00:13:11, the subtitles are mixed up and some appears late and for not enough time. Especially the following sentences: 
"Philippe qui buzzait depuis tout à  l'heure car il avait la bonne réponse."

**What did you expect to happen?**
The subtitles to be readable properly

**What actually happened?**
It seems the region nor the timing are handled correctly. Working with DashJS though.

**Comment by theodab** [comment]
Yeah, it definitely looks like something is going wrong. I'll look into it.

**Comment by theodab** [comment]
As far as the timing issue goes, it looks like there are multiple unrelated problems here.

1. Examining your TTML more closely, some of the files seem to have multiple XML headers (`<?xml>` tags). It looks like you are, in some cases, concatenating two TTML files together or something?
This is invalid XML, and thus invalid TTML. Because of this, `DOMParser.parseFromString` was ignoring the rest of the file. This was causing some of the captions to be ignored. It looks like DASHJS uses imscJS for TTML parsing, so I guess imscJS must be more lenient about bad syntax.
My recent CL makes Shaka Player error in this case; we were not catching those errors, and they should rightfully be passed on to the user.
It would be possible to modify Shaka Player's TTML parser to handle this case. But I don't think that it would be a good idea to support invalid input, unless it's a very common mistake. So this might be something for you to fix on your end.
1. There's some problem that sometimes causes captions to not display unless the mouse happens to be moving near the video element. This only happens when playing at full speed, and only on Chrome, so I think it's some kind of rendering engine bug. We will bring this up to some Chrome people to figure out.

As far as I could tell, there weren't any actual timing errors; besides those two issues, I didn't notice any timing differences between Shaka Player and DASHJS.

The regions, on the other hand, are in fact totally being ignored. Our custom UI text displayer just never implemented that feature, it looks like. I filed a separate issue for this, #2191, since it's a missing feature rather than a bug.

**Comment by happypoulp** [comment]
Hi,

Thanks for your feedback, we'll investigate to confirm what produces those multiple XML headers and if it should be considered an issue (it's not quite clear as of now what the TTML specs says about it). We'll get back to you as soon as we have more infos.

**Comment by theodab** [comment]
I've improved how Shaka Player handles TTML with regions, in a [recent commit](https://github.com/google/shaka-player/commit/2b1ffa36485a0f1bc10d8f73ba6dfbb0e206d43c). I wasn't able to find much sample content for TTML with regions, and when examining how DASH.js treats them I found at least one bug. So I'm not actually sure if this new behavior is what you expect or not, in terms of appearance.

Here's a screenshot:
![Screen Shot 2019-11-14 at 3 39 01 PM](https://user-images.githubusercontent.com/21047092/68905120-f32fbb80-06f4-11ea-81ec-94b19838bab5.png)
If this isn't how the captions are supposed to look, feel free re-open issue #2191 and complain about that.

Oh, and the problem where the captions don't display if the mouse isn't moving nearby seems to have gone away? I guess one of the CSS changes must have inadvertently fixed it. So all that's left should be the problem where multiple XML headers end up in the output.
